### PR TITLE
Admin/API Role + User creation functionality added

### PR DESCRIPTION
**Pull request by furan917**
Added the ability to create new Admin User Roles.

Yaml file takes name of Role and a list of Resource IDs to create role.
List of all available ResourceIds added for lookup
If user already exists and resources not empty, user resources will be updated.

### CODE REVIEW

File: Model/Component/AdminRoles.php [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+        $this->roleFactory = $roleFactory;
+        $this->rulesFactory = $rulesFactory;
+    }
+
+    /**
+     * @param array $data
+     * @SuppressWarnings(PHPMD)
+     */
+    protected function processData($data = null)
```
Comment (adragus-inviqa) [inline Model/Component/AdminRoles.php]: Should be `processData(array $data = null)`, amirite?

File: Model/Component/AdminRoles.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        }
+    }
+
+    /**
+     * Create Admin user roles, or update them if they exist
+     *
+     * @param $roleName
+     * @param $resources
+     */
+    private function createAdminRole($roleName, $resources)
```
Comment (adragus-inviqa) [inline Model/Component/AdminRoles.php]: Maybe add the types of `$roleName` and `$resources` in phpdoc and the signature, too, if you can? I think for `$resources` you can, as that's an array.

File: Model/Component/AdminRoles.php [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        $this->setResourceIds($role, $resources);
+    }
+
+    /**
+     * Set ResourceIDs the Admin Role will have access to
+     *
+     * @param role
+     * @param array|null $resources
+     */
+    private function setResourceIds($role, $resources)
```
Comment (adragus-inviqa) [inline Model/Component/AdminRoles.php]: Please add the role type in phpdoc and maybe change the 2nd param to `array $resources = null`?

File: Model/Component/AdminUsers.php [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+        $this->userFactory = $userFactory;
+        $this->roleFactory = $roleFactory;
+    }
+
+    /**
+     * @param array $data
+     * @SuppressWarnings(PHPMD)
+     */
+    protected function processData($data = null)
```
Comment (adragus-inviqa) [inline Model/Component/AdminUsers.php]: `array $data = null` :P

**Comment by furan917** [comment]
Improvements made

**Pull request merged**